File: Backend API/Locafest/com.locafest.Domain/LocacaoContext/Entities/Veiculo.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace com.locafest.Domain.LocacaoContext.Entities
{
    public class Veiculo : BaseEntity
    {
        public string Placa { get; set; }
        public int Ano { get; set; }
        public decimal ValorHora { get; set; }
        public int LimitePortaMala { get; set; }
        public Guid CategoriaId { get; set; }
        public Guid CombustivelId { get; set; }
        public Guid MarcaId { get; set; }
        public Guid ModeloId { get; set; }

        public Marca Marca { get; set; }
        public Modelo Modelo { get; set; }
    }
}

File: Mobile FrontEnd/locafest/lib/app/shared/entities/veiculo.dart
class Veiculo {
  String id;
  String placa;
  int ano;
  int valorHora;
  int limitePortaMala;
  String categoriaId;
  String combustivelId;
  String marcaId;
  String modeloId;

  Veiculo(
      {this.id,
      this.placa,
      this.ano,
      this.valorHora,
      this.limitePortaMala,
      this.categoriaId,
      this.combustivelId,
      this.marcaId,
      this.modeloId});

  Veiculo.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    placa = json['placa'];
    ano = json['ano'];
    valorHora = json['valorHora'];
    limitePortaMala = json['limitePortaMala'];
    categoriaId = json['categoriaId'];
    combustivelId = json['combustivelId'];
    marcaId = json['marcaId'];
    modeloId = json['modeloId'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['placa'] = this.placa;
    data['ano'] = this.ano;
    data['valorHora'] = this.valorHora;
    data['limitePortaMala'] = this.limitePortaMala;
    data['categoriaId'] = this.categoriaId;
    data['combustivelId'] = this.combustivelId;
    data['marcaId'] = this.marcaId;
    data['modeloId'] = this.modeloId;
    return data;
  }
}